otwcode/otwarchive
# AO3-3582 Update jquery.livequery.min.js to v1.3.6

# Pull Request Checklist

  ^ autocomplete already seems to be thoroughly tested in the cucumber tests; the show / hide links are also tested in cucumber tests.

## Issue

https://otwarchive.atlassian.net/browse/AO3-3582

## Purpose

Updates the version of the jquery livequery plugin to 1.3.6 (from 1.1.1) to be slightly more up-to-date.

## Testing Instructions

Because this is a version upgrade, I believe it can only be tested by verifying that the relevant functionality is still working. As far as I know this package is now only used for autocomplete, so to test, you can go to a page that uses an autocomplete widget and walk through a workflow that uses autocomplete. For example:

1. Go to Search > Edit Your Search
2. Fill out a fandom in the search field
3. Submit the search form.

Livequery was also being used to make scrolling work correctly for comment pagination and for the comments link at the top of a work.

1. Go to a work with more than one page of comments.
2. Click on the Comments link at the top of the page. You should be scrolled down to the comments section and comments should display.
3. Click on a pagination link at the bottom of the comments list. You should be scrolled back up to the top of the comments list.

## Additional information

The version of 1.3.6 from https://plugins.jquery.com/livequery/ wasn't working right for me so I got the version from https://github.com/hazzik/livequery/tree/1.3.6 (the repository that was linked as the homepage for the project in the plugin registry.)

I had to make a couple of updates to application.js because there was lingering usage of livequery to capture live events on elements. This usage was removed from livequery in v1.2 because it was supplanted by jquery.live, which was removed in jquery 1.9 (the current version in use by AO3) in favor of [event delegation with `.on()`](https://api.jquery.com/on/)

I'm not sure if there's an appropriate way to test that the scrolling functionality the changed callbacks represent is working properly in cucumber, but there are already tests for the other behavior associated with these links.

## Credit

*What name do you want us to use to credit your work in the [Archive of Our Own's Release Notes](https://archiveofourown.org/admin_posts?tag=1)?*

Stephen Burrows

*What pronouns do you prefer (she/her, he/him, zie/hir etc)?*
he/him
